cpp/src/plasma/client.cc

```diff
@@ -500,7 +503,11 @@ Status PlasmaClient::Impl::GetBuffers(
             data + object->data_offset, object->data_size + object->metadata_size);
       } else {
 #ifdef PLASMA_CUDA
-        physical_buf = gpu_object_map.find(object_ids[i])->second->ptr;
+        std::lock_guard<std::mutex> lock(gpu_mutex);
+        auto iter = gpu_object_map.find(object_ids[i]);
+        ARROW_CHECK(iter != gpu_object_map.end());
```

Comment:
It seems this item can get deleted in another concurrent local client, why the CHECK?

Reply:
This CHECK will always be TRUE because "object_entry == objects_in_use_.end()" is false.
If you think the CHECK is redundant, I can remove it.

Reply:
@fsaintjacques Excuse me, do you have any more questions and need I remove the CHECK?

Reply:
Nope this is good for me.